File: go-sdk/pkg/models/operations/listdeployoperations.go
package operations

type ListDeployOperationsPathParams struct {
	Database     string `pathParam:"style=simple,explode=false,name=database"`
	Number       string `pathParam:"style=simple,explode=false,name=number"`
	Organization string `pathParam:"style=simple,explode=false,name=organization"`
}

type ListDeployOperationsQueryParams struct {
	Page    *float64 `queryParam:"style=form,explode=true,name=page"`
	PerPage *float64 `queryParam:"style=form,explode=true,name=per_page"`
}

type ListDeployOperationsRequest struct {
	PathParams  ListDeployOperationsPathParams
	QueryParams ListDeployOperationsQueryParams
}

type ListDeployOperationsResponse struct {
	ContentType                                  string
	StatusCode                                   int64
	ListDeployOperations200ApplicationJSONObject map[string]map[string]interface{}
}

File: ts-sdk/src/sdk/models/operations/listdeployoperations.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import { Type } from "class-transformer";


export class ListDeployOperationsPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=database" })
  database: string;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=number" })
  number: string;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=organization" })
  organization: string;
}

export class ListDeployOperationsQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=page" })
  page?: number;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=per_page" })
  perPage?: number;
}

export class ListDeployOperationsRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: ListDeployOperationsPathParams;

  @SpeakeasyMetadata()
  queryParams: ListDeployOperationsQueryParams;
}

export class ListDeployOperationsResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  listDeployOperations200ApplicationJSONObject?: Record<string, Record<string, any>>;
}